src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -299,6 +305,11 @@ private void GatherMiningStatistics()
 
             this.miningStatistics.MinerHits = hitCount;
 
+            if (this.signals != null)
```

Comment:
Any reason for the null check?

Reply:
If there is no signals, I guess it may throw error publishing the event.